extensions/wikia/AdEngine/src/templates/big-fancy-ad-above-desktop-config.js

```diff
@@ -1,54 +1,89 @@
-				const isInViewport = isElementInViewport(adSlot, params);
-
-				pinNavbar(isInViewport && !isSticky);
-				this.moveNavbar(isSticky ? container.offsetHeight : 0);
-			};
-
-			this.slotConfig = params.config;
-			wrapper.style.opacity = '0';
-
-			slotTweaker.onReady(adSlot).then(() => {
-				wrapper.style.opacity = '';
-				updateNavbar();
+const {
+	CSS_CLASSNAME_FADE_IN_ANIMATION,
+	CSS_CLASSNAME_SLIDE_OUT_ANIMATION,
+	CSS_CLASSNAME_STICKY_BFAA,
+	CSS_TIMING_EASE_IN_CUBIC
+} = universalAdPackage;
+
+export const getConfig = () => ({
+	adSlot: null,
+	slotParams: null,
+	updateNavbarOnScroll: null,
+
+	slotsToEnable: [
+		'BOTTOM_LEADERBOARD',
+		'INCONTENT_BOXAD_1'
+	],
+
+	onInit(adSlot, params) {
+		this.adSlot = adSlot;
+		this.slotParams = params;
+		context.set(`slots.${adSlot.getSlotName()}.options.isVideoMegaEnabled`, params.isVideoMegaEnabled);
+
+		const spotlightFooter = document.getElementById('SPOTLIGHT_FOOTER');
+		const wrapper = document.getElementById('WikiaTopAds');
+
+		wrapper.style.opacity = '0';
+		slotTweaker.onReady(adSlot).then(() => {
+			wrapper.style.opacity = '';
+			this.updateNavbar();
+		});
+		this.updateNavbarOnScroll = scrollListener.addCallback(() => this.updateNavbar());
+
+		if (!window.ads.runtime.disableCommunitySkinOverride) {
+			document.body.classList.add('uap-skin');
+		}
+
+		if (spotlightFooter) {
+			spotlightFooter.parentNode.style.display = 'none';
+		}
+	},
+
+	onStickBfaaCallback(adSlot) {
+		adSlot.getElement().classList.add(CSS_CLASSNAME_STICKY_BFAA);
+		pinNavbar(false);
+	},
+
+	onUnstickBfaaCallback(adSlot) {
+		const slideOutDuration = 600;
+
+		scrollListener.removeCallback(this.updateNavbarOnScroll);
+		this.updateNavbarOnScroll = null;
+		Object.assign(navBarElement.style, {
+			transition: `top ${slideOutDuration}ms ${CSS_TIMING_EASE_IN_CUBIC}`,
+			top: '0'
+		});
+		animateUAPSlot(adSlot, CSS_CLASSNAME_SLIDE_OUT_ANIMATION, slideOutDuration).then(() => {
+			Object.assign(navBarElement.style, {
+				transition: '',
+				top: ''
 			});
-			scrollListener.addCallback(updateNavbar);
-
-			if (!window.ads.runtime.disableCommunitySkinOverride) {
-				document.body.classList.add('uap-skin');
-			}
-			if (spotlightFooter) {
-				spotlightFooter.parentNode.style.display = 'none';
-			}
-		},
-		onStickBfaaCallback(adSlot) {
-			adSlot.getElement().classList.add('sticky-bfaa');
-			pinNavbar(false);
-		},
-		onUnstickBfaaCallback(adSlot, params) {
-			adSlot.getElement().classList.remove('sticky-bfaa');
-			pinNavbar(isElementInViewport(adSlot, params));
-		},
-		moveNavbar(offset) {
-			if (navBarElement) {
-				navBarElement.style.top = offset ? `${offset}px` : '';
-			}
+			adSlot.getElement().classList.remove(CSS_CLASSNAME_STICKY_BFAA);
+			this.updateNavbar();
+
+			return animateUAPSlot(adSlot, CSS_CLASSNAME_FADE_IN_ANIMATION, 400);
+		}).then(() => {
+			this.updateNavbarOnScroll = scrollListener.addCallback(() => this.updateNavbar());
```

Comment:
Do we need to register scroll callback again after we completed BFAA exit animation? I believe headroom should handle the navbar from this moment.